CDCgov/prime-reportstream
# HL7 OBX Mapping: handle multiple segments

This PR adds handling for multiple `OBX` segments for `OML` messages. It was missed from our previous PR that added OBX mapping

Test Steps:
1. Run `./gradlew primeCLI --args='fhirdata --output-format HL7 --input-file "./src/testIntegration/resources/datatests/FHIR_to_HL7/sample_OML_20230831-0001.fhir"  -s "./metadata/hl7_mapping/receivers/Flexion/TILabOrder.yml"'`
2. Check that the output HL7 message has the `OBX` segment more than once

## Changes
- Added `resultIndex` to hl7 path to handle multiple segments
- Added additional observation segment to sample file in order to test multiple OBX

## Checklist

### Testing

## Linked Issues
[Create Outbound OML OBX Segment](https://github.com/CDCgov/trusted-intermediary/issues/495)
